### PR TITLE
Detect API use errors earlier.

### PR DESCRIPTION
**Pull request by swankjesse**
Previously we would accept invalid input, but crash when that invalid input
was used.

### CODE REVIEW

File: okhttp-android-support/src/main/java/okhttp3/internal/huc/JavaApiConverter.java [hunk 1]
```diff
@@ -648,6 +648,9 @@ public String getHeaderField(int position) {
       if (position == 0) {
         return StatusLine.get(response).toString();
       }
+      if (position > response.headers().size()) {
```
Comment (swankjesse) [inline okhttp-android-support/src/main/java/okhttp3/internal/huc/JavaApiConverter.java]: OkHttp gets strict, but HttpURLConnection stays loose!

File: okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 import java.net.URL;
 import java.net.URLConnection;
 import java.net.UnknownHostException;
+import java.security.KeyManagementException;
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java]: Unused

File: okhttp/src/main/java/okhttp3/Headers.java [hunk 3]
```diff
@@ -75,20 +75,12 @@ public int size() {
...
   public String name(int index) {
-    int nameIndex = index * 2;
-    if (nameIndex < 0 || nameIndex >= namesAndValues.length) {
-      return null;
-    }
-    return namesAndValues[nameIndex];
+    return namesAndValues[index * 2];
   }
 
   /** Returns the value at {@code index} or null if that is out of range. */
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Headers.java]: Doc needs updated.

File: okhttp/src/main/java/okhttp3/Headers.java [hunk 4]
```diff
@@ -75,20 +75,12 @@ public int size() {
 
   /** Returns the field at {@code position} or null if that is out of range. */
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Headers.java]: Doc needs updated.

**Comment by JakeWharton** [comment]
LGTM

File: okhttp/src/main/java/okhttp3/Headers.java (hunk 3 above)
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Headers.java]: Good good. Fixed

**Pull request merged**